### PR TITLE
Add support for custom CA cert

### PR DESCRIPTION
**Pull request by gordonbondon**
Add support for custom ca certs and insecure connections when Elastic has a self-signed certificate.

### CODE REVIEW

**Comment by phillbaker** [comment]
Thanks for the PR! Can you also update the example in the README with these configuration examples?

File: provider.go [hunk 1]
```diff
@@ -72,6 +89,8 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	if m := awsUrlRegexp.FindStringSubmatch(parsedUrl.Hostname()); m != nil {
 		log.Printf("[INFO] Using AWS: %+v", m[1])
 		opts = append(opts, elastic.SetHttpClient(awsHttpClient(m[1], d)), elastic.SetSniff(false))
+	} else {
+		opts = append(opts, elastic.SetHttpClient(tlsHttpClient(d)), elastic.SetSniff(false))
```
Comment (phillbaker) [inline provider.go]: Would we always want to create this http client even if we don't have a cacert_file? Should this be `else if (d.Get("insecure") || d.Get("cacert_file"))` ?

Comment (gordonbondon) [reply provider.go]: Sounds good. I'll update my PR.

Comment (phillbaker) [reply provider.go]: Thanks, don't forget to squash please!

Comment (gordonbondon) [reply provider.go]: Done!

**Comment by phillbaker** [comment]
Thanks!

**Pull request merged**